### PR TITLE
Implements Overlap function

### PR DESCRIPTION
**Pull request by lyriccoder**
Implements Overlap function in according  to the passage of the paper:
min_overlap

Paper:
Identifying Extract Method Refactoring Opportunities Based on Functional Relevance

### CODE REVIEW

File: veniq/__init__.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+"""
+Veniq uses Machine Learning to analyze source code, find possible refactorings, and suggest those that seem optimal
+"""
+
+__version__ = '0.0.0'
+__author__ = 'Yegor Bugayenko'
```
Comment (KatGarmash) [inline veniq/__init__.py:6]: why Yegor Bugayenko?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# The MIT License (MIT)
```
Comment (KatGarmash) [inline setup.py:1]: shouldn't be in a separate PR?

File: veniq/baselines/semi/overlap.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+from itertools import combinations
+from typing import List
+import operator
+
+MIN_OVERLAP = 0.1
```
Comment (KatGarmash) [inline veniq/baselines/semi/overlap.py:29]: min_overlap should be a paramter of is_overlap

File: veniq/baselines/semi/overlap.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    :param a: array of 2 elements, the first is start statements number,
+    the second is end statements number
+    :param b: array of 2 elements, the first is start statements number,
+    the second is end statements number
+
+    :return: True if it is an overlap, False if it is not
+    """
+
+    a_LoC = abs(a[1] - a[0])
```
Comment (KatGarmash) [inline veniq/baselines/semi/overlap.py:44]: we should check/agree on the convention of how we represent ranges of lines: whether the end line is included or excluded

File: veniq/baselines/semi/overlap.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    b_LoC = abs(b[1] - b[0])
+    overlap = len(set(range(a[0], a[1] + 1)) & set(range(b[0], b[1] + 1)))
+    overlap_index = overlap / float(max(a_LoC, b_LoC))
+    if overlap_index >= MIN_OVERLAP:
```
Comment (KatGarmash) [inline veniq/baselines/semi/overlap.py:48]: can be simplified to "return  overlap_index >= MIN_OVERLAP"

**Review by KatGarmash — CHANGES_REQUESTED** [review]
see comments

File: veniq/__init__.py (hunk 1 above)
Comment (lyriccoder) [reply veniq/__init__.py:6]: cause he is the main person, the same thing was in aibolit

Comment (acheshkov) [reply veniq/__init__.py:6]: Please change it )

Comment (lyriccoder) [reply veniq/__init__.py:6]: What should we write here?

File: veniq/baselines/semi/overlap.py (hunk 5 above)
Comment (lyriccoder) [reply veniq/baselines/semi/overlap.py]: fixed

File: veniq/baselines/semi/overlap.py (hunk 3 above)
Comment (lyriccoder) [reply veniq/baselines/semi/overlap.py]: fixed

File: setup.py (hunk 2 above)
Comment (lyriccoder) [reply setup.py:1]: deleted

File: veniq/__init__.py (hunk 1 above)
Comment (lyriccoder) [reply veniq/__init__.py]: fixed

File: veniq/baselines/semi/overlap.py (hunk 4 above)
Comment (lyriccoder) [reply veniq/baselines/semi/overlap.py]: fixed

File: veniq/__init__.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+"""
+Veniq uses Machine Learning to analyze source code, find possible refactorings, and suggest those that seem optimal
```
[the remark this answers is not in the document]
Comment (lyriccoder) [reply veniq/__init__.py]: fixed

File: veniq/baselines/semi/overlap.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
+from typing import List
+
+
+def is_overlap(a: List[int], b: List[int], min_overlap=0.1):
```
Comment (acheshkov) [inline veniq/baselines/semi/overlap.py:4]:
```suggestion
def is_overlap(a: Tuple[int, int], b: Tuple[int, int], min_overlap: float=0.1) -> bool:
```

**Review by acheshkov — CHANGES_REQUESTED** [review]

**Review by acheshkov — APPROVED** [review]

**Review by KatGarmash — APPROVED** [review]
